replica.go

```diff
@@ -221,8 +226,20 @@ func (r *Replica) syncWAL(ctx context.Context) (err error) {
 		return err
 	})
 
+	var ew io.WriteCloser = pw
+
+	// Add encryption if we have recipients.
+	if len(r.AgeRecipients) > 0 {
+		var err error
+		ew, err = age.Encrypt(pw, r.AgeRecipients...)
+		defer ew.Close()
+		if err != nil {
+			return err
+		}
```

Comment:
The `defer ew.Close()` needs to be moved after the `if` block. If an error returns then `ew` will be `nil` and panic.